tests/StrictPhpTest/TypeFinder/ReturnTypeFinderTest.php

```diff
@@ -0,0 +1,160 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license.
+ */
+
+namespace StrictPhpTest\TypeFinder;
+
+use Go\Aop\Framework\DynamicReflectionMethodInvocation;
+use StrictPhp\TypeChecker\ApplyTypeChecks;
+use StrictPhp\TypeChecker\TypeChecker;
+use StrictPhp\TypeFinder\ReturnTypeFinder;
+use StrictPhpTestAsset\ClassWithReturnTypeMethod;
+
+/**
+ * Tests for {@see \StrictPhp\TypeFinder\ReturnTypeFinder}
+ *
+ * @author Jefersson Nathan <malukenho@phpse.net>
+ * @license MIT
+ *
+ * @group Coverage
+ *
+ * @covers \StrictPhp\TypeFinder\ReturnTypeFinder
+ */
+class ReturnTypeFinderTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var ReturnTypeFinder
+     */
+    private $finder;
+
+    public function setUp()
+    {
+        $typeCheckers = [
+            new TypeChecker\IntegerTypeChecker(),
+            new TypeChecker\CallableTypeChecker(),
+            new TypeChecker\StringTypeChecker(),
+            new TypeChecker\GenericObjectTypeChecker(),
+            new TypeChecker\ObjectTypeChecker(),
+            new TypeChecker\MixedTypeChecker(),
+            new TypeChecker\BooleanTypeChecker(),
+            new TypeChecker\NullTypeChecker(),
+        ];
+
+        $typeCheckers[] = new TypeChecker\TypedTraversableChecker(...$typeCheckers);
+
+        $applyTypeChecks = new ApplyTypeChecks(...$typeCheckers);
+
+        $this->finder = new ReturnTypeFinder($applyTypeChecks);
+    }
+
+    /**
+     * @dataProvider mixedAnnotationTypes
+     *
+     * @param $class
+     * @param $methodName
+     * @param $params
+     */
+    public function testRetrievesMethodReturnTypes($class, $methodName, $params)
+    {
+        $finder = $this->finder;
```

Comment:
@TomasVotruba `callable` does not guarantee existence of the `__invoke()` method